src/lib/actionUtils/actionUtils.ts

```diff
@@ -82,6 +88,10 @@ export function getShortcutHandlersMap(
 
 type ActiveParentActionIdListArg = Readonly<Array<ActionId>>;
 
+export function getParentAction(action: WrappedAction, actions: DeepReadonly<Actions>) {
+  return Object.values(actions).filter(({id}) => id === action.parentActionId)[0]
```

Comment:
chore: this will cause each child action to walk through the list and find its parent action. So far we have tried to avoid this. We only walk from parent to child because that way in one cycle we have all items.

I think for now we should just have a prop at the Root component level that do you want only top level actions or do you want all actions to be listed from the start. Let's not give this option at individual action level.

Reply:
To clarify, is this what you're proposing?

- remove `isolateChildren` from `Action`, to avoid the need for per-child parent lookup
- add `initialVisibleActions` to `Root`, to apply the functionality to the entire palette

Reply:
exactly

Reply:
resolved @ [`5d95e4f` (#15)](https://github.com/itaditya/solid-command-palette/pull/15/commits/5d95e4f136ff94325e547076cb44f50adfce7fac)